src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -472,7 +493,33 @@ public MethodIL GetMethodILWithInlinedSubstitutions(MethodIL method)
                 debugInfo = new SubstitutedDebugInformation(debugInfo, sequencePoints.ToArray());
             }
 
-            return new SubstitutedMethodIL(method, newBody, newEHRegions.ToArray(), debugInfo);
+            ArrayBuilder<string> newStrings = default;
+            if (hasGetResourceStringCall)
+            {
+                for (int offset = 0; offset < flags.Length; offset++)
+                {
+                    if ((flags[offset] & OpcodeFlags.GetResourceStringCall) == 0)
+                        continue;
+
+                    Debug.Assert(newBody[offset] == (byte)ILOpcode.call);
+                    var getter = (EcmaMethod)method.GetObject(new ILReader(newBody, offset + 1).ReadILToken());
+
+                    string resourceString = GetResourceStringForAccessor(getter);
+                    if (resourceString == null)
+                        continue;
+
+                    int tokenValue = NewStringTokenRidStart - newStrings.Count;
```

Comment:
Might need to add backout logic if we ran out of string tokens. Hypothetically we could. It's likely not very real (and would be totally untested).